pkg/storegateway/bucket.go

```diff
@@ -1274,7 +1274,7 @@ func (s *BucketStore) openBlocksForReading(ctx context.Context, skipChunks bool,
 
 	indexReaders := make(map[ulid.ULID]*bucketIndexReader, len(blocks))
 	for _, b := range blocks {
-		indexReaders[b.meta.ULID] = b.loadedIndexReader(s.postingsStrategy, stats)
+		indexReaders[b.meta.ULID] = b.loadedIndexReader(spanCtx, s.postingsStrategy, stats)
 	}
 	if skipChunks {
 		return blocks, indexReaders, nil
```

Comment:
Note to reviewers: I've deliberately used `ctx` in the call to `chunkReader` below as that ensures that spans created by the chunk reader are attached to the caller of `openBlocksForReading`, rather than the span created by `openBlocksForReading`.